### PR TITLE
FLPATH-319 - Enable integration test via CI

### PR DESCRIPTION
**Pull request by gciavarrini**
Documentation in [Parodos docs](https://github.com/parodos-dev/parodos-dev.github.io/pull/11)

With this change, `run_examples.sh` is *NOT* executed anymore from the CI. 
Future new test scenarios must be part of `integration-tests` package.

Generate issues:
- https://issues.redhat.com/browse/FLPATH-348. 
- https://issues.redhat.com/browse/FLPATH-394

They will be fixed in other PRs.

Co-authored-by: Gloria Ciavarrini <gciavarrini@redhat.com>
Co-authored-by: Moti Asayag <masayag@redhat.com>

### CODE REVIEW

File: integration-tests/pom.xml [hunk 1]
```diff
@@ -113,4 +113,44 @@
...
+                                <goals>
+                                    <goal>integration-test</goal>
+                                    <goal>verify</goal>
+                                </goals>
+                            </execution>
+                        </executions>
+                        <configuration>
+                            <includes>
+                                <include>**/*IntegrationTest.java</include>
+                                <include>**/*Flow.java</include>
```
Comment (masayag) [inline integration-tests/pom.xml:144]: I think we should change this to FlowTest.java and rename the test classes accordingly.

**Comment by masayag** [comment]
Shouldn't there be additional changes after the async-execution PR was merged?

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -180,12 +190,13 @@ jobs:
           }
           kubectl get pods --all-namespaces
 
-      - name: Run example script
+      - name: Maven Build
+        run: mvn -Dmaven.test.skip=true clean install
```
Comment (eloycoto) [inline .github/workflows/test.yml:194]: If we use clean here you're not using the cache at all, so or delete the cache or remove the clean.

File: integration-tests/pom.xml (hunk 1 above)
Comment (gciavarrini) [reply integration-tests/pom.xml]: Addressed

File: .github/workflows/test.yml (hunk 2 above)
Comment (gciavarrini) [reply .github/workflows/test.yml]: Thanks! `clean` removed

File: integration-tests/pom.xml [hunk 3]
```diff
@@ -113,4 +113,44 @@
             </plugin>
         </plugins>
     </build>
+    <profiles>
+        <!--
+         Integration test profile.
+         Explicitly triggered with -P integrationtest or by environment variable "integrationtest" set
+         -->
+        <profile>
+            <id>integrationtest</id>
```
[the remark this answers is not in the document]
Comment (gciavarrini) [reply integration-tests/pom.xml]: Done!

File: workflow-examples/src/main/java/com/redhat/parodos/examples/prebuilt/PrebuiltWorkFlowConfiguration.java [hunk 4]
```diff
@@ -1,22 +1,30 @@
...
 @Configuration
+@Slf4j
 public class PrebuiltWorkFlowConfiguration {
 
 	@Bean
 	NotificationWorkFlowTask notificationTask() {
-		return new NotificationWorkFlowTask("http://localhost:8081",
+		String serverIp = Optional.ofNullable(System.getenv("NOTIFICATION_SERVER_ADDRESS")).orElse("localhost");
+		String serverPort = Optional.ofNullable(System.getenv("SERVER_PORT")).orElse("8080");
+		String notificationWorkFlowBasePath = "http://" + serverIp + ":" + serverPort;
```
Comment (masayag) [inline workflow-examples/src/main/java/com/redhat/parodos/examples/prebuilt/PrebuiltWorkFlowConfiguration.java:25]: just for the symmetry, let's name this one: NOTIFICATION_SERVER_PORT

**Review by pkliczewski — APPROVED** [review]

**Pull request merged**